internal/certgen/certgen.go

```diff
@@ -326,3 +487,48 @@ func bigIntHash(n *big.Int) []byte {
 	h.Write(n.Bytes())
 	return h.Sum(nil)
 }
+
+func serviceNames(service, namespace string) []string {
+	return []string{
+		service,
+		fmt.Sprintf("%s.%s", service, namespace),
+		fmt.Sprintf("%s.%s.svc", service, namespace),
+		fmt.Sprintf("%s.%s.svc.cluster.local", service, namespace),
+	}
+}
+
+func writePEMFile(filename string, data []byte) error {
+	_, err := os.Stat(filename)
+	if err != nil {
+		// Can't stat the file, so we'll create it
+		err = ioutil.WriteFile(filename, data, 0644)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("Created %s\n", filename)
+		return nil
+	}
+	return fmt.Errorf("can't overwrite %s", filename)
+
+}
+
+func writeSecret(filename string, secret *corev1.Secret) error {
+	s := json.NewYAMLSerializer(json.DefaultMetaFactory, scheme.Scheme, scheme.Scheme)
+	_, err := os.Stat(filename)
+	if err != nil {
+		// Can't stat the file, we'll create it
+		f, err := os.Create(filename)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
```

Comment:
Sadly you cannot use defer here as we're writing to f and operating systems may not report an error to write until we try to close (which implicitly flushes the file).